articles/implementing-aws-s3-functionalities-on-a-nodejs-graphql-api/index.md

```diff
@@ -0,0 +1,624 @@
+
+- In the redirected page, search for s3 in the search bar and click the first result.
+
+- You will be redirected to your s3 account page.
+
+- On the top right, there is your username and a dropdown arrow. Click here and then click `My security credentials`.
+
+- On the new page, click on `Access keys (access key ID and secret access key)`.
+
+- If you already have an `Access Key ID` and a `Secret access key` feel free to use them. But if you don't have one, click the `Create New Access Key` button.
+
+- Download the `Key File`.
+
+- From the current URL, you pick your region. For example, considering this URL: `https://console.aws.amazon.com/iam/home?region=us-east-2#/security_credentials`, the region is set to us-east-2.
+
+- In the `Key File` you downloaded, add the following:
+
+```javascript
+region=//the region from the URL. e.g us-east-2
+```
+
+- After setting the region, you are good for the next step.
+
+### Setting up the development server
+
+To set up the development server, clone this [Github repository](https://github.com/mwangiKibui/aws-sdk-node.js-s3-graphql). In the repository, in the `src` directory, there is the start and final folder. Throughout the article, we will be working on the start folder. In case you encounter an error or you want to compare your code, feel free to check the final folder.
+
+From the `Key File` you downloaded copy the data to the `.env` file appropriately. Ensure you match the names so that there is no conflict.
+
+In the cloned folder, run the following command to install the necessary dependencies:
+
+```bash
+npm install
+```
+
+Proceed to the start folder and go through the schema files to understand how we are structuring the data. Our focus throughout the article will be on the resolvers folder. It is therefore helpful to have a grasp of the underlying schema.
+
+### Creating an s3 bucket
+
+An s3 bucket is the foundational storage instance in s3. It consists of folders and objects. The objects are the files. Data in the bucket is stored as either a folder or an object.
+
+To create a bucket, we implement the following functionality in the `resolvers/mutation-resolvers.js` under `createBucket()` function:
+
+```javascript
+//create a bucket.
+async createBucket(bucketName){
+
+    const params = {
+        Bucket:bucketName
+    };
+
+    let create_bucket = promisify(this.s3.createBucket.bind(this.s3));
+
+    await create_bucket(params).catch(console.log);
+
+    return {
+        success:true,
+        message:"Bucket created successfully."
+    };
+
+};
+```
+
+From the above implementation:
+
+- Create params object with the key `Bucket` to hold the sent `bucketName`.
+
+- Promisify the `createBucket()` function from s3 so that we can use *async/await* syntax.
+
+- Call the `createBucket()` function passing along the params object.
+
+- Send back a response matching the schema.
+
+To test this:
+
+- Start your development server by running the following command from your terminal:
```

Comment:
Please also mention that the user must open the terminal in the current working directory and provide instructions for the same.

Reply:
fixed